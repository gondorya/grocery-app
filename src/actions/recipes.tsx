import uuid from 'uuid';

const addRecipe = (
	{ title = '', description = '', steps = [], ingredients = [], date = new Date(), img = '' } = {}
) => {
	return {
		type: 'ADD_RECIPE',
		recipe: {
			id: uuid(),
			title,
			description,
			steps,
			ingredients,
			img,
			date
		}
	};
};

const removeRecipe = ({ id = '' } = {}) => {
	return {
		type: 'REMOVE_RECIPE',
		id
	};
};

const editRecipe = (id, updates) => {
	return {
		type: 'EDIT_RECIPE',
		id,
		updates
	};
};

export { addRecipe, removeRecipe, editRecipe };
