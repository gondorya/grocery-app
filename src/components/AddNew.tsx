import React from 'react';

const AddNew = () => <p>Add New</p>;

export default AddNew;
