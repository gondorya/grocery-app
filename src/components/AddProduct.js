import Button from './atoms/Button';

const AddProduct = () => (
	<form>
		<input type="text" name="product" placeholder="new product" />
		<Button className="btn--submit" text="Add new product" type="submit" />
	</form>
);

export default AddProduct;
