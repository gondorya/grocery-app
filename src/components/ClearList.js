import React from 'react';
import Button from './atoms/Button';

const ClearList = (props) => (
	<div>
		<Button onClick={props.handleClick} className="btn--reset" text="Clear List" />
	</div>
);

export default ClearList;
