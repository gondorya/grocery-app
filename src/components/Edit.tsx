import React from 'react';

const Edit = (props) => (
	<div>
		<p>Page with id {props.match.params.id}</p>
	</div>
);

export default Edit;
