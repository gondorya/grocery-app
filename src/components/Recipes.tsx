import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { filterRecipes } from '../selectors/selectors';
import RecipeListItem from './RecipeListItem';
import RecipesFilter from './RecipesFilter';

interface Recipe {
	id: number;
	name: string;
}

const Recipes = (props) => (
	<div>
		<Link to="recipes/add-recipe" className="btn btn--primary btn--toRight">
			Add recipe
		</Link>
		<RecipesFilter />
		<ul className="recipeList">{props.recipes.map((recipe) => <RecipeListItem key={recipe.id} {...recipe} />)}</ul>
	</div>
);

const mapStateToProps = (state) => {
	return {
		recipes: filterRecipes(state.recipes, state.filters)
	};
};

export default connect(mapStateToProps)(Recipes);
