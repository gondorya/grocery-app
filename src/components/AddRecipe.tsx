import React from 'react';
import { connect } from 'react-redux';
import RecipeForm from './RecipeForm';
import { addRecipe } from '../actions/recipes';

class AddRecipe extends React.Component {
	constructor(props) {
		super(props);
	}

	newRecipe = (recipe) => {
		this.props.newRecipe(recipe);
		this.props.history.push('/recipes');
	};

	render() {
		return (
			<div>
				<h2>Add New Recipe</h2>
				<RecipeForm newRecipe={this.newRecipe} />
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch) => ({ newRecipe: (recipe) => dispatch(addRecipe(recipe)) });

export default connect(undefined, mapDispatchToProps)(AddRecipe);
