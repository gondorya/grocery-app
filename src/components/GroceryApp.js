import React from 'react';
import Products from './Products';
import AddProduct from './AddProduct';
import ClearList from './ClearList';

class GroceryApp extends React.Component {
	state = {
		products: []
	};

	componentDidMount = () => {
		const data = JSON.parse(localStorage.getItem('products'));

		this.setState(() => ({
			products: data ? data : []
		}));
	};

	componentDidUpdate = () => {
		localStorage.setItem('products', JSON.stringify(this.state.products));
	};

	handleSubmit = (newProduct) => {
		this.setState((prevState) => ({
			products: prevState.products.concat(newProduct)
		}));
	};

	updateProducts = (newList) => {
		this.setState(() => ({
			products: newList
		}));
	};

	resetList = (e) => {
		e.preventDefault;
		this.setState(() => ({
			products: []
		}));
	};

	render() {
		return (
			<div>
				<AddProduct
					handleSubmit={this.handleSubmit}
					products={this.state.products}
					updateProducts={this.updateProducts}
					changeModalState={this.changeModalState}
				/>
				<Products
					products={this.state.products}
					updateProducts={this.updateProducts}
					handleDelete={this.deleteProduct}
				/>
        <ClearList handleClick={this.resetList} />
			</div>
		);
	}
}

export default GroceryApp;
