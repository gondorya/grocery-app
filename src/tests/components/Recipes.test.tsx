import React from 'react';
import { shallow } from 'enzyme';
import { Recipes } from '../../components/Recipes';
import recipes from '../fixtures/recipes';

test('should render list of recipes', () => {
	const wrapper = shallow(<Recipes recipes={recipes} />);
	expect(wrapper).toMatchSnapshot();
});
