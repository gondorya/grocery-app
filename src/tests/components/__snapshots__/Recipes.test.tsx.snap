// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of recipes 1`] = `
<div>
  <Link
    className="btn btn--primary btn--toRight"
    to="recipes/add-recipe"
  >
    Add recipe
  </Link>
  <Connect(RecipesFilter) />
  <ul
    className="recipeList"
  >
    <RecipeListItem
      date="2018-09-15T00:00:00+02:00"
      description="english street food"
      id="1"
      img=""
      ingredients={
        Array [
          "fish",
          "chips",
          "peas",
        ]
      }
      key="1"
      steps={
        Array [
          "step 1",
          "step 2",
          "step 3",
        ]
      }
      title="fish and chips"
    />
    <RecipeListItem
      date={2017-09-19T22:00:00.000Z}
      description="vegan salad with fresh vegetabels"
      id="2"
      img=""
      ingredients={
        Array [
          "salad",
          "tomato",
          "onion",
          "olive",
        ]
      }
      key="2"
      steps={
        Array [
          "step 1",
          "step 2",
          "step 3",
        ]
      }
      title="salad"
    />
    <RecipeListItem
      date={2018-12-09T23:00:00.000Z}
      description="Perfect with chips"
      id="3"
      img=""
      ingredients={
        Array [
          "salad",
          "tomato",
          "onion",
          "olive",
        ]
      }
      key="3"
      steps={
        Array [
          "step 1",
          "step 2",
          "step 3",
        ]
      }
      title="Hanburger"
    />
  </ul>
</div>
`;
